internal/pkg/checks/pathRulesCheck.go

```diff
@@ -135,4 +241,102 @@ var pathRulesBarSubpathCheck = &Check{
 	},
 }
 
-// TODO: Implement more checks on edge cases like leading `/`, query params, and encoding
+var pathRulesPrefixAaaBbbcccCheck = &Check{
+	Name:        "path-rules-prefix-aaa-bbbccc",
+	Description: "Ingress with prefix path rule with a trailing slash should not match string prefix (/aaa/bbb/ does not match /aaa/bbbccc)",
+	Run: func(check *Check, config Config) (success bool, err error) {
+		req, res, err := captureRequest(fmt.Sprintf("http://%s/aaa/bbbccc", pathRulesHost), "path-rules")
+		if err != nil {
+			return
+		}
+
+		a := new(AssertionSet)
+		// Assert the request received from the downstream service
+		a.Equals(req.DownstreamServiceId, "path-rules-catchall", "expected the downstream service would be '%s' but was '%s'")
+		a.Equals(req.Path, "/aaa/bbbccc", "expected the request path would be '%s' but was '%s'")
+		// Assert the downstream service response
+		a.Equals(res.StatusCode, 200, "expected statuscode to be %s but was %s")
+
+		if a.Error() == "" {
+			success = true
+		} else {
+			fmt.Print(a)
```

Comment:
I think you should return AssertionSet, otherwise every user will need to remember to print it out.

Reply:
This was captured in https://github.com/kubernetes-sigs/ingress-controller-conformance/issues/18